### DISCUSSION TITLE
How can i update a container image for a lambda function

### DISCUSSION THREAD

**Discussion by ChrisDAT20** [aws/aws-cli#7182] [Q&A] [answered]
Hello,

I created a container image to be able to overcome the size limits per lambda function (250mb). 
The container image has been created, but the Python code for the lambda_handler needs some fixes. 
How can i update the container image?
The Dockerfile contains a command which pulls the Pythoncode from Gitlab and installs all necessary dependencies.
sam build reports it takes everything from cache, and when I use --no-cached, sam build still takes everything from cache.

Or, maybe I am completely wrong on how to update lambda functions and its dependencies, like the container image.

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @ChrisDAT20, thanks for reaching out.

Have you tried using `update-function-code` AWS CLI command to update Lambda function code with container image type? You can use `--image-uri (string)` option as mentioned in the docs below.

> URI of a container image in the Amazon ECR registry. Do not use for a function defined with a .zip file archive.

Here's link to the documentation: https://docs.aws.amazon.com/cli/latest/reference/lambda/update-function-code.html

Hope it helps but please let me know if you have any other questions!
